aeron-archive/src/main/cpp_wrapper/client/archive/ArchiveContext.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2014-2024 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef AERON_ARCHIVE_WRAPPER_CONTEXT_H
+#define AERON_ARCHIVE_WRAPPER_CONTEXT_H
+
+#include "AeronArchive.h"
+#include "CredentialsSupplier.h"
+#include "concurrent/YieldingIdleStrategy.h"
+
+#include "Context.h"
```

Comment:
I'd love for the aeron::archive::client::Context to be defined in a file called 'Context.h'.  But if I do that, then I have no way of including the Context.h from the base client cpp_wrapper project.  I'm not sure how to fix that... or even if it matters.